shared/update_check_settings_file.cpp

```diff
@@ -0,0 +1,103 @@
+#include "update_check_settings_file.h"
+#include <version/eps_version.h>
+#include <helper/resource_paths_juce-file.hpp>
+
+UpdateCheckerSettingsFile::UpdateCheckerSettingsFile()
+{
+    settingsFile = ResourcePaths::getSettingsDirectory(true).getChildFile("UpdateCheck.settings");
+    if (!settingsFileExists()) {
+        // No settings file - perhaps first run?
+        if (saveSettings() && settingsFileExists()) {
```

Comment:
I try to factor this sort of thing into an
```c++
bool ensureSettingsFileExists() {
  bool success = settingsFileExists();
  if (!success) {
     success = saveSettings() && settingsFileExists();
  }
  return success;
``` 
as it makes the resulting `if(ensureSettingsFileExists())` more clear at a glance that a negative return is probably something untoward (permission error or whatever) rather than having to trace through nested `if`s.

If it fails what should the extension do?